maven-plugin/pom.xml

```diff
@@ -96,11 +96,11 @@
     <dependency>
       <groupId>${project.groupId}</groupId>
       <artifactId>wildfly-feature-spec-gen</artifactId>
-      <scope>provided</scope>
```

Comment:
Did it create a problem?

Reply:
Yes, if you develop the new version it would not be picked up properly. It would stick to the "provided" old version.

The alternative would be jinxing the build to provide the version under development.

Ultimately it is a problem in Maven because it doesn't allow for proper dependency management on plugins.

Reply:
I need to test but it might be there for a reason. There is forked (default) and not forked mode. There is some classpath setup, I recall I had an issue with deps being automatically added to the classpath which I wanted to avoid.
The version of this artifact can be controlled by explicitly referencing it from the project's pom.xml. At least it worked when WildFly Core release used an earlier version and Full switched to the latest one.

Reply:
Current WildFly master will fail to build the servlet feature-pack with this change.

Reply:
I changed it back to provided for now.

Reply:
You should read the failure:
```
[ERROR] Failed to execute goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:3.0.0.Alpha2-SNAPSHOT:generate-feature-specs (feature-spec-build) on project wildfly-servlet-galleon-pack: Execution feature-spec-build of goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:3.0.0.Alpha2-SNAPSHOT:generate-feature-specs failed: A required class was missing while executing org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:3.0.0.Alpha2-SNAPSHOT:generate-feature-specs: org/wildfly/core/embedded/EmbeddedManagedProcess
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:3.0.0.Alpha2-SNAPSHOT
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/carlo/.m2/repository/org/wildfly/galleon-plugins/wildfly-galleon-maven-plugin/3.0.0.Alpha2-SNAPSHOT/wildfly-galleon-maven-plugin-3.0.0.Alpha2-SNAPSHOT.jar
[ERROR] urls[1] = file:/home/carlo/.m2/repository/org/wildfly/galleon-plugins/wildfly-galleon-plugins/3.0.0.Alpha2-SNAPSHOT/wildfly-galleon-plugins-3.0.0.Alpha2-SNAPSHOT.jar
[ERROR] urls[2] = file:/home/carlo/.m2/repository/org/jboss/galleon/galleon-core/3.0.0.Alpha1/galleon-core-3.0.0.Alpha1.jar
[ERROR] urls[3] = file:/home/carlo/.m2/repository/org/jboss/staxmapper/1.1.0.Final/staxmapper-1.1.0.Final.jar
[ERROR] urls[4] = file:/home/carlo/.m2/repository/com/googlecode/java-diff-utils/diffutils/1.3.0/diffutils-1.3.0.jar
[ERROR] urls[5] = file:/home/carlo/.m2/repository/org/jboss/galleon/galleon-maven-plugin/3.0.0.Alpha1/galleon-maven-plugin-3.0.0.Alpha1.jar
[ERROR] urls[6] = file:/home/carlo/.m2/repository/org/eclipse/aether/aether-util/1.1.0/aether-util-1.1.0.jar
[ERROR] urls[7] = file:/home/carlo/.m2/repository/org/apache/maven/plugin-tools/maven-plugin-annotations/3.5.2/maven-plugin-annotations-3.5.2.jar
[ERROR] urls[8] = file:/home/carlo/.m2/repository/org/jboss/galleon/galleon-maven-universe/3.0.0.Alpha1/galleon-maven-universe-3.0.0.Alpha1.jar
[ERROR] urls[9] = file:/home/carlo/.m2/repository/org/wildfly/galleon-plugins/wildfly-feature-spec-gen/3.0.0.Alpha2-SNAPSHOT/wildfly-feature-spec-gen-3.0.0.Alpha2-SNAPSHOT.jar
[ERROR] urls[10] = file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-filtering/3.1.1/maven-filtering-3.1.1.jar
[ERROR] urls[11] = file:/home/carlo/.m2/repository/org/apache/maven/maven-builder-support/3.3.1/maven-builder-support-3.3.1.jar
[ERROR] urls[12] = file:/home/carlo/.m2/repository/javax/enterprise/cdi-api/1.0/cdi-api-1.0.jar
[ERROR] urls[13] = file:/home/carlo/.m2/repository/javax/annotation/jsr250-api/1.0/jsr250-api-1.0.jar
[ERROR] urls[14] = file:/home/carlo/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.inject/0.3.0/org.eclipse.sisu.inject-0.3.0.jar
[ERROR] urls[15] = file:/home/carlo/.m2/repository/org/sonatype/sisu/sisu-guice/3.2.5/sisu-guice-3.2.5-no_aop.jar
[ERROR] urls[16] = file:/home/carlo/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
[ERROR] urls[17] = file:/home/carlo/.m2/repository/com/google/guava/guava/16.0.1/guava-16.0.1.jar
[ERROR] urls[18] = file:/home/carlo/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[ERROR] urls[19] = file:/home/carlo/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[20] = file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.0.0/maven-shared-utils-3.0.0.jar
[ERROR] urls[21] = file:/home/carlo/.m2/repository/com/google/code/findbugs/jsr305/2.0.1/jsr305-2.0.1.jar
[ERROR] urls[22] = file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-utils/3.1.0/plexus-utils-3.1.0.jar
[ERROR] urls[23] = file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.22/plexus-interpolation-1.22.jar
[ERROR] urls[24] = file:/home/carlo/.m2/repository/org/sonatype/plexus/plexus-build-api/0.0.7/plexus-build-api-0.0.7.jar
[ERROR] urls[25] = file:/home/carlo/.m2/repository/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5.jar
[ERROR] urls[26] = file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-archiver/3.4/plexus-archiver-3.4.jar
[ERROR] urls[27] = file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-io/2.7.1/plexus-io-2.7.1.jar
[ERROR] urls[28] = file:/home/carlo/.m2/repository/commons-io/commons-io/2.5/commons-io-2.5.jar
[ERROR] urls[29] = file:/home/carlo/.m2/repository/org/apache/commons/commons-compress/1.11/commons-compress-1.11.jar
[ERROR] urls[30] = file:/home/carlo/.m2/repository/org/iq80/snappy/snappy/0.4/snappy-0.4.jar
[ERROR] urls[31] = file:/home/carlo/.m2/repository/org/tukaani/xz/1.5/xz-1.5.jar
[ERROR] urls[32] = file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-artifact-transfer/0.9.1/maven-artifact-transfer-0.9.1.jar
[ERROR] urls[33] = file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.7.1/plexus-component-annotations-1.7.1.jar
[ERROR] urls[34] = file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-common-artifact-filters/3.0.1/maven-common-artifact-filters-3.0.1.jar
[ERROR] urls[35] = file:/home/carlo/.m2/repository/org/sonatype/sisu/sisu-inject-bean/1.4.2/sisu-inject-bean-1.4.2.jar
[ERROR] urls[36] = file:/home/carlo/.m2/repository/org/sonatype/sisu/sisu-guice/2.1.7/sisu-guice-2.1.7-noaop.jar
[ERROR] urls[37] = file:/home/carlo/.m2/repository/commons-codec/commons-codec/1.6/commons-codec-1.6.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------: org.wildfly.core.embedded.EmbeddedManagedProcess
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :wildfly-servlet-galleon-pack
```
It fails because it can't find wildfly-core-embedded which is supposed to be provided.

Reply:
https://github.com/wolfc/jboss-eap7/commit/d505c24a1ca75b81ae1b0458c22a172c46cfde04#diff-600376dffeb79835ede4a0b285078036R7039

Reply:
I just said that if you revert the change, it will work.

Reply:
Like I said, it does not:
```
[DEBUG] Embedded server classpath:
[DEBUG]  org.codehaus.plexus.classworlds.realm.ClassRealm
[DEBUG]   1. file:/home/carlo/.m2/repository/org/wildfly/galleon-plugins/wildfly-galleon-maven-plugin/2.0.1.Alpha1-SNAPSHOT/wildfly-galleon-maven-plugin-2.0.1.Alpha1-SNAPSHOT.jar
[DEBUG]   2. file:/home/carlo/.m2/repository/org/wildfly/galleon-plugins/wildfly-galleon-plugins/2.0.1.Alpha1-SNAPSHOT/wildfly-galleon-plugins-2.0.1.Alpha1-SNAPSHOT.jar
[DEBUG]   3. file:/home/carlo/.m2/repository/org/jboss/galleon/galleon-core/2.0.0.Final/galleon-core-2.0.0.Final.jar
[DEBUG]   4. file:/home/carlo/.m2/repository/org/jboss/staxmapper/1.1.0.Final/staxmapper-1.1.0.Final.jar
[DEBUG]   5. file:/home/carlo/.m2/repository/com/googlecode/java-diff-utils/diffutils/1.3.0/diffutils-1.3.0.jar
[DEBUG]   6. file:/home/carlo/.m2/repository/org/jboss/galleon/galleon-maven-plugin/2.0.0.Final/galleon-maven-plugin-2.0.0.Final.jar
[DEBUG]   7. file:/home/carlo/.m2/repository/org/eclipse/aether/aether-util/1.1.0/aether-util-1.1.0.jar
[DEBUG]   8. file:/home/carlo/.m2/repository/org/apache/maven/plugin-tools/maven-plugin-annotations/3.5.2/maven-plugin-annotations-3.5.2.jar
[DEBUG]   9. file:/home/carlo/.m2/repository/org/jboss/galleon/galleon-maven-universe/2.0.0.Final/galleon-maven-universe-2.0.0.Final.jar
[DEBUG]   10. file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-filtering/3.1.1/maven-filtering-3.1.1.jar
[DEBUG]   11. file:/home/carlo/.m2/repository/org/apache/maven/maven-builder-support/3.3.1/maven-builder-support-3.3.1.jar
[DEBUG]   12. file:/home/carlo/.m2/repository/javax/enterprise/cdi-api/1.0/cdi-api-1.0.jar
[DEBUG]   13. file:/home/carlo/.m2/repository/javax/annotation/jsr250-api/1.0/jsr250-api-1.0.jar
[DEBUG]   14. file:/home/carlo/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.inject/0.3.0/org.eclipse.sisu.inject-0.3.0.jar
[DEBUG]   15. file:/home/carlo/.m2/repository/org/sonatype/sisu/sisu-guice/3.2.5/sisu-guice-3.2.5-no_aop.jar
[DEBUG]   16. file:/home/carlo/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
[DEBUG]   17. file:/home/carlo/.m2/repository/com/google/guava/guava/16.0.1/guava-16.0.1.jar
[DEBUG]   18. file:/home/carlo/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[DEBUG]   19. file:/home/carlo/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[DEBUG]   20. file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.0.0/maven-shared-utils-3.0.0.jar
[DEBUG]   21. file:/home/carlo/.m2/repository/com/google/code/findbugs/jsr305/2.0.1/jsr305-2.0.1.jar
[DEBUG]   22. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-utils/3.1.0/plexus-utils-3.1.0.jar
[DEBUG]   23. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.22/plexus-interpolation-1.22.jar
[DEBUG]   24. file:/home/carlo/.m2/repository/org/sonatype/plexus/plexus-build-api/0.0.7/plexus-build-api-0.0.7.jar
[DEBUG]   25. file:/home/carlo/.m2/repository/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5.jar
[DEBUG]   26. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-archiver/3.4/plexus-archiver-3.4.jar
[DEBUG]   27. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-io/2.7.1/plexus-io-2.7.1.jar
[DEBUG]   28. file:/home/carlo/.m2/repository/commons-io/commons-io/2.5/commons-io-2.5.jar
[DEBUG]   29. file:/home/carlo/.m2/repository/org/apache/commons/commons-compress/1.11/commons-compress-1.11.jar
[DEBUG]   30. file:/home/carlo/.m2/repository/org/iq80/snappy/snappy/0.4/snappy-0.4.jar
[DEBUG]   31. file:/home/carlo/.m2/repository/org/tukaani/xz/1.5/xz-1.5.jar
[DEBUG]   32. file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-artifact-transfer/0.9.1/maven-artifact-transfer-0.9.1.jar
[DEBUG]   33. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.7.1/plexus-component-annotations-1.7.1.jar
[DEBUG]   34. file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-common-artifact-filters/3.0.1/maven-common-artifact-filters-3.0.1.jar
[DEBUG]   35. file:/home/carlo/.m2/repository/org/sonatype/sisu/sisu-inject-bean/1.4.2/sisu-inject-bean-1.4.2.jar
[DEBUG]   36. file:/home/carlo/.m2/repository/org/sonatype/sisu/sisu-guice/2.1.7/sisu-guice-2.1.7-noaop.jar
[DEBUG]   37. file:/home/carlo/.m2/repository/commons-codec/commons-codec/1.6/commons-codec-1.6.jar
[DEBUG]  java.net.URLClassLoader
[DEBUG]   1. file:/home/carlo/.m2/repository/org/wildfly/galleon-plugins/wildfly-feature-spec-gen/2.0.0.Final/wildfly-feature-spec-gen-2.0.0.Final.jar
[DEBUG]   2. file:/tmp/jboss-eap7/servlet-galleon-pack/target/wildfly/jboss-modules.jar
[DEBUG]   3. file:/home/carlo/.m2/repository/org/wildfly/core/wildfly-cli/6.0.7.Final-redhat-00001/wildfly-cli-6.0.7.Final-redhat-00001-client.jar
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:2.0.1.Alpha1-SNAPSHOT:generate-feature-specs (feature-spec-build) on project wildfly-servlet-galleon-pack: Feature spec generator failed: Failed to generate feature specs: org.wildfly.galleon.plugin.featurespec.generator.FeatureSpecGenerator.<init>(java.lang.String, java.nio.file.Path, java.util.Map, boolean, boolean) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:2.0.1.Alpha1-SNAPSHOT:generate-feature-specs (feature-spec-build) on project wildfly-servlet-galleon-pack: Feature spec generator failed
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:50)
Caused by: org.apache.maven.plugin.MojoExecutionException: Feature spec generator failed
	at org.wildfly.galleon.maven.WfFeatureSpecBuildMojo.doExecute(WfFeatureSpecBuildMojo.java:302)
	at org.wildfly.galleon.maven.WfFeatureSpecBuildMojo.execute(WfFeatureSpecBuildMojo.java:191)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 27 more
Caused by: org.jboss.galleon.ProvisioningException: Failed to generate feature specs
	at org.wildfly.galleon.maven.FeatureSpecGeneratorInvoker.generateSpecs(FeatureSpecGeneratorInvoker.java:55)
	at org.wildfly.galleon.maven.WfFeatureSpecBuildMojo.doExecute(WfFeatureSpecBuildMojo.java:300)
	... 30 more
Caused by: java.lang.NoSuchMethodException: org.wildfly.galleon.plugin.featurespec.generator.FeatureSpecGenerator.<init>(java.lang.String, java.nio.file.Path, java.util.Map, boolean, boolean)
	at java.lang.Class.getConstructor0(Class.java:3082)
	at java.lang.Class.getConstructor(Class.java:1825)
	at org.wildfly.galleon.maven.FeatureSpecGeneratorInvoker.generateSpecs(FeatureSpecGeneratorInvoker.java:46)
	... 31 more
```

Reply:
That change hides the problem. It still leaves wildfly-feature-spec-gen in the wrong class loader.
```
[DEBUG] Embedded server classpath:
[DEBUG]  org.codehaus.plexus.classworlds.realm.ClassRealm
[DEBUG]   1. file:/home/carlo/.m2/repository/org/wildfly/galleon-plugins/wildfly-galleon-maven-plugin/2.0.2.Alpha1-SNAPSHOT/wildfly-galleon-maven-plugin-2.0.2.Alpha1-SNAPSHOT.jar
[DEBUG]   2. file:/home/carlo/.m2/repository/org/wildfly/galleon-plugins/wildfly-galleon-plugins/2.0.2.Alpha1-SNAPSHOT/wildfly-galleon-plugins-2.0.2.Alpha1-SNAPSHOT.jar
[DEBUG]   3. file:/home/carlo/.m2/repository/org/jboss/galleon/galleon-core/2.0.1.Final/galleon-core-2.0.1.Final.jar
[DEBUG]   4. file:/home/carlo/.m2/repository/org/jboss/staxmapper/1.1.0.Final/staxmapper-1.1.0.Final.jar
[DEBUG]   5. file:/home/carlo/.m2/repository/com/googlecode/java-diff-utils/diffutils/1.3.0/diffutils-1.3.0.jar
[DEBUG]   6. file:/home/carlo/.m2/repository/org/jboss/galleon/galleon-maven-plugin/2.0.1.Final/galleon-maven-plugin-2.0.1.Final.jar
[DEBUG]   7. file:/home/carlo/.m2/repository/org/eclipse/aether/aether-util/1.1.0/aether-util-1.1.0.jar
[DEBUG]   8. file:/home/carlo/.m2/repository/org/apache/maven/plugin-tools/maven-plugin-annotations/3.5.2/maven-plugin-annotations-3.5.2.jar
[DEBUG]   9. file:/home/carlo/.m2/repository/org/jboss/galleon/galleon-maven-universe/2.0.1.Final/galleon-maven-universe-2.0.1.Final.jar
[DEBUG]   10. file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-filtering/3.1.1/maven-filtering-3.1.1.jar
[DEBUG]   11. file:/home/carlo/.m2/repository/org/apache/maven/maven-builder-support/3.3.1/maven-builder-support-3.3.1.jar
[DEBUG]   12. file:/home/carlo/.m2/repository/javax/enterprise/cdi-api/1.0/cdi-api-1.0.jar
[DEBUG]   13. file:/home/carlo/.m2/repository/javax/annotation/jsr250-api/1.0/jsr250-api-1.0.jar
[DEBUG]   14. file:/home/carlo/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.inject/0.3.0/org.eclipse.sisu.inject-0.3.0.jar
[DEBUG]   15. file:/home/carlo/.m2/repository/org/sonatype/sisu/sisu-guice/3.2.5/sisu-guice-3.2.5-no_aop.jar
[DEBUG]   16. file:/home/carlo/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
[DEBUG]   17. file:/home/carlo/.m2/repository/com/google/guava/guava/16.0.1/guava-16.0.1.jar
[DEBUG]   18. file:/home/carlo/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[DEBUG]   19. file:/home/carlo/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[DEBUG]   20. file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.0.0/maven-shared-utils-3.0.0.jar
[DEBUG]   21. file:/home/carlo/.m2/repository/com/google/code/findbugs/jsr305/2.0.1/jsr305-2.0.1.jar
[DEBUG]   22. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-utils/3.1.0/plexus-utils-3.1.0.jar
[DEBUG]   23. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.22/plexus-interpolation-1.22.jar
[DEBUG]   24. file:/home/carlo/.m2/repository/org/sonatype/plexus/plexus-build-api/0.0.7/plexus-build-api-0.0.7.jar
[DEBUG]   25. file:/home/carlo/.m2/repository/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5.jar
[DEBUG]   26. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-archiver/3.4/plexus-archiver-3.4.jar
[DEBUG]   27. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-io/2.7.1/plexus-io-2.7.1.jar
[DEBUG]   28. file:/home/carlo/.m2/repository/commons-io/commons-io/2.5/commons-io-2.5.jar
[DEBUG]   29. file:/home/carlo/.m2/repository/org/apache/commons/commons-compress/1.11/commons-compress-1.11.jar
[DEBUG]   30. file:/home/carlo/.m2/repository/org/iq80/snappy/snappy/0.4/snappy-0.4.jar
[DEBUG]   31. file:/home/carlo/.m2/repository/org/tukaani/xz/1.5/xz-1.5.jar
[DEBUG]   32. file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-artifact-transfer/0.9.1/maven-artifact-transfer-0.9.1.jar
[DEBUG]   33. file:/home/carlo/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.7.1/plexus-component-annotations-1.7.1.jar
[DEBUG]   34. file:/home/carlo/.m2/repository/org/apache/maven/shared/maven-common-artifact-filters/3.0.1/maven-common-artifact-filters-3.0.1.jar
[DEBUG]   35. file:/home/carlo/.m2/repository/org/sonatype/sisu/sisu-inject-bean/1.4.2/sisu-inject-bean-1.4.2.jar
[DEBUG]   36. file:/home/carlo/.m2/repository/org/sonatype/sisu/sisu-guice/2.1.7/sisu-guice-2.1.7-noaop.jar
[DEBUG]   37. file:/home/carlo/.m2/repository/commons-codec/commons-codec/1.6/commons-codec-1.6.jar
[DEBUG]  java.net.URLClassLoader
[DEBUG]   1. file:/home/carlo/.m2/repository/org/wildfly/galleon-plugins/wildfly-feature-spec-gen/2.0.2.Alpha1-SNAPSHOT/wildfly-feature-spec-gen-2.0.2.Alpha1-SNAPSHOT.jar
[DEBUG]   2. file:/home/carlo/work/github/wildfly/wildfly/servlet-galleon-pack/target/wildfly/jboss-modules.jar
[DEBUG]   3. file:/home/carlo/.m2/repository/org/wildfly/core/wildfly-cli/7.0.0.Alpha2/wildfly-cli-7.0.0.Alpha2-client.jar
```

Reply:
I tried commenting out 'provided' for the spec-gen and the embedded and adding them to the plugin dependencies. It still failed to build. Do you have it working on your branch? I want to test forking and not forking. If that works, I'll probably be ok with those changes.